src/libasr/pass/pass_list_concat.cpp

```diff
@@ -0,0 +1,114 @@
+#include <libasr/asr.h>
+#include <libasr/containers.h>
+#include <libasr/exception.h>
+#include <libasr/asr_utils.h>
+#include <libasr/asr_verify.h>
+#include <libasr/pass/pass_list_concat.h>
+#include <libasr/pass/pass_utils.h>
+
+#include <vector>
+#include <utility>
+
+
+namespace LFortran {
+
+using ASR::down_cast;
+using ASR::is_a;
+
+/*
+This ASR pass replaces list-concat to
+
+Converts:
+    x = i + j // lists
+
+to:
+
+ */
+
+
+class ReplaceListConcat : public ASR::BaseExprReplacer<ReplaceListConcat>
+{
+    private:
+
+        Allocator& al;
+        ASR::TranslationUnit_t &unit;
+
+    public:
+        ReplaceListConcat(Allocator &al_, ASR::TranslationUnit_t &unit_) :
+        al(al_), unit(unit_)
+        { }
+
+        void create_concat_function() {
+            SymbolTable *fn_scope = al.make_new<SymbolTable>(unit.m_global_scope);
+            std::string fn_name = "_lcompilers_list_concat";
+            ASR::ttype_t *type;
+            Location loc;
+            ASR::asr_t *return_var=nullptr;
+            ASR::expr_t *return_var_ref=nullptr;
+            char *var_name;
+            int idx = 1;
+
+            ASR::asr_t *fn = ASR::make_Function_t(
+            al, loc,
+            /* a_symtab */ fn_scope,
+            /* a_name */ (char *)fn_name.c_str(),
+            /* a_args */ nullptr,
+            /* n_args */ 0,
+            /* a_body */ nullptr,
+            /* n_body */ 0,
+            /* a_return_var */ return_var_ref,
+            ASR::abiType::BindC,
+            ASR::Public, ASR::Implementation,
+            nullptr,
+            false, false, false, false, false,
+            nullptr, 0,
+            nullptr, 0,
+            false);
+            unit.m_global_scope->add_symbol(fn_name, down_cast<ASR::symbol_t>(fn));
+        }
```

Comment:
> I will generate the code for `_lcompilers_list_concat` using ASR APIs

@czgdp1807, you meant to do something like this, right?

Reply:
Or 
```py
def _lcompilers_list_concat(left: list[i32], right: list[i32]) -> list[i32]:
    l: list[i32] = []
    i: i32
    for i in left:
        l.append(i)
    for i in right:
        l.append(i)
    return l
```
Add the above to function to `src/runtime/lpython_builtin.py`(or something similar) and call that func in this pass